src/ripple/basics/Buffer.h

```diff
@@ -40,8 +40,29 @@ class Buffer
 
 public:
     Buffer() = default;
-    Buffer (Buffer const&) = delete;
-    Buffer& operator= (Buffer const&) = delete;
+
+    Buffer& operator= (Buffer const& other)
+    {
+        size_ = other.size_;
+        if (size_)
+        {
+            p_.reset (new std::uint8_t[size_]);
```

Comment:
Exception unsafe - if `new` fails here, `size_` will contain the wrong information and the object will be left in an inconsistent statements.